infra/scripts/setup-common-functions.sh

```diff
@@ -10,14 +10,17 @@ install_test_tools() {
 }
 
 install_gcloud_sdk() {
-  print_banner "Installing Google Cloud SDK"
+  print_banner "Installing Google Cloud SDK. (GOOGLE_APPLICATION_CREDENTIALS: '$GOOGLE_APPLICATION_CREDENTIALS')_"
   if [[ ! $(command -v gsutil) ]]; then
     CURRENT_DIR=$(dirname "$BASH_SOURCE")
-    . "${CURRENT_DIR}"/install-google-cloud-sdk.sh
+    # Activate service account credentials if specified.
+    if [ -z $GOOGLE_APPLICATION_CREDENTIALS ]
+    then
+      . "${CURRENT_DIR}"/install-google-cloud-sdk.sh
+    else
+      . "${CURRENT_DIR}"/install-google-cloud-sdk.sh --with-key-file $GOOGLE_APPLICATION_CREDENTIALS
```

Comment:
Is there a reason to keep that installation script separate and not move it into this functions script? This just seems like a lot of brittle scripts